app/controllers/diary_entry_controller.rb

```diff
@@ -57,10 +61,17 @@ def comment
     @diary_comment = @entry.comments.build(comment_params)
     @diary_comment.user = @user
     if @diary_comment.save
-      if @diary_comment.user != @entry.user
-        Notifier.diary_comment_notification(@diary_comment).deliver_now
+
+      # Notify current subscribers of the new comment
+      @entry.subscribers.visible.each do |user|
+        if @user != user
```

Comment:
Well the two are the same as `@diary_comment.user` has just been set from `@user` so it doesn't really matter too much.